core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -116,6 +118,19 @@ private void initializeCatalogTables() throws InterruptedException, SQLException
       LOG.debug("Creating table {} to store iceberg catalog", JdbcUtil.CATALOG_TABLE_NAME);
       return conn.prepareStatement(JdbcUtil.CREATE_CATALOG_TABLE).execute();
     });
+
+    connections.run(conn -> {
+      DatabaseMetaData dbMeta = conn.getMetaData();
+      ResultSet tableExists = dbMeta.getTables(null, null, JdbcUtil.NAMESPACE_PROPERTIES_TABLE_NAME, null);
```

Comment:
Nit: Consider either wrapping this with a utility method or adding inline comments where the nulls are for what they represent, like `dbMeta.getTables(null /* ??? */, null …. )`.

I’d personally use the named function approach. This many nullls (or Boolean literals, etc) makes it hard for the reader to grok the function call’s signature and therefore what it’s doing. In this case, because you have a good variable name, it’s not as difficult to tell what the intended code is. But it’s still arguably pretty difficult to look at this and visually assert that this is likely correct or what all of the unused parameters are.


Given that this code path isn’t exactly hit by any means, I’d go with the utility function to wrap this. It could even just return the boolean, as the ResultSet is otherwise unused.

Reply:
It was specified the same way for the earlier table. I think naming parameters makes more sense. Adding in the next commit.

Reply:
@rdblue  any thoughts on this. I looked at both utility and comments. The utility function needs a connection to be passed as well to return the ResultSet
```
DatabaseMetaData dbMeta = conn.getMetaData();
      ResultSet tableExists = dbMeta.getTables(null, null, JdbcUtil.CATALOG_TABLE_NAME, null);

```
and does this look like 
```
dbMeta.getTables(null /** catalog name **/, null /** schemaPattern **/, JdbcUtil.CATALOG_TABLE_NAME /** tableNamePattern **/ ,  null /** types **/); 
```
IDEs do give the method signature and searching the method should be trivial.

Reply:
I think that @kbendick's recommendation is good in general, but probably not required here. Probably preferred, which is why it was just a nit.

Also, I'd caution against relying on an IDE to make code readable. While IDEs may help, we do a lot of code reviews in github where we don't have the benefit of automatic comments, and even in IntelliJ I have to look up what it is instead of just reading.

Reply:
Makes sense. Fixed this.